src/libPMacc/include/particles/memory/dataTypes/Particle.hpp

```diff
@@ -110,6 +111,12 @@ struct Particle : public InheritLinearly<typename T_FrameType::MethodsList>
     >::type
     operator[](const T_Key key)
     {
+        PMACC_CASSERT_MSG_TYPE(
+            key_not_exists_in_particle,
+            T_Key,
+            traits::HasIdentifier< ThisType, T_Key >::type::value
```

Comment:
Use `Particle` instead of `ThisType`. Defining `ThisType` in the first place is a common beginner error and should be removed.